samples/camera/camerax/src/main/java/com/example/platform/camerax/mlkit/QrCodeViewModel.kt

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2025 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.platform.camerax.mlkit
+
+import android.content.Intent
+import android.graphics.Rect
+import android.view.MotionEvent
+import android.view.View
+import androidx.core.net.toUri
+import com.google.mlkit.vision.barcode.common.Barcode
+
+/**
+ * A ViewModel for encapsulating the data for a QR Code, including the encoded data, the bounding
+ * box, and the touch behavior on the QR Code.
+ *
+ * As is, this class only handles displaying the QR Code data if it's a URL. Other data types
+ * can be handled by adding more cases of Barcode.TYPE_URL in the init block.
+ */
+class QrCodeViewModel(barcode: Barcode) {
```

Comment:
I'm not sure overall if this class name is adapted as it doesn't follow the usual logic of a view model on Android and a reference to the context shouldn't be extracted from the event listener but CameraX/MLKit API constraints make it hard to follow the Android way...